libconnman-qt/sessionagent.cpp

```diff
@@ -23,12 +23,15 @@ This class is used to run a connman session.
 
   */
 
-SessionAgent::SessionAgent(const QString &path, QObject* parent) :
-    QObject(parent),
-    agentPath(path),
-    m_manager(NetworkManagerFactory::createInstance()),
-    m_session(0)
+SessionAgent::SessionAgent(const QString &path, QObject *parent)
+    : QObject(parent)
+    , agentPath(path)
+    , m_manager(NetworkManager::sharedInstance())
+    , m_session(nullptr)
 {
+    // FIXME: the session mode is long deprecated property on connman, and this
```

Comment:
It is not deprecated in ConnMan, we just do not use it.

Reply:
connman/doc/manager-api.txt:

```
                boolean SessionMode [readwrite]  [experminental][deprecated]

                        This property exists only for compatibility reasons
                        and does not affect ConnMan in any way.

                        The default value is false.
```

Changed by:
commit 0d69def2fda8c9a365b1114a7d5dba7350461035
Author: Patrik Flykt <patrik.flykt@linux.intel.com>
Date:   Thu Aug 29 15:06:25 2013 +0300

Now I just added this quick comment, but thinking if I should remove these call altogether, make the NetworkManager setter either a deprecated no-op or hide the whole thing. Same consideration for QML side.

Reply:
Ah yes the particular parameter is. Oh, right. Well since we don't use session I hadn't really much paid attention to that, heh.